### DISCUSSION TITLE
Add crontab syntax highlighting despite only 4K downloads on packagecontrol.io

### DISCUSSION THREAD

**Discussion by KaKi87** [sharkdp/bat#2424] [Ideas]
Hello,

I would like to request an exception to the `10,000 downloads on packagecontrol.io` syntax request rule for [Crontab](https://packagecontrol.io/packages/Crontab) *(which has 4K downloads there)* because it is Linux's task scheduler.

Then, it would be usable with `crontab -l | bat -l crontab`.

Thanks

**Comment by keith-hall** [comment]
I would agree to this exemption. It would be nice if the language hint would be unnecessary to help it be more discoverable that `bat` supports it but I guess that is unrealistic.

**Reply by sharkdp** [reply]
I also agree. Thank you for creating the request.

**Reply by KaKi87** [reply]
Maybe a child pipe process can get the ID of the parent pipe process :shrug:

**Reply by sharkdp** [reply]
I had that some thought years ago (and asked this question on S.E.: https://unix.stackexchange.com/questions/405485/name-of-the-process-on-the-other-end-of-a-unix-pipe). There is a way to do that. I even wrote a Rust program ("[magic](https://gist.github.com/sharkdp/b1273e71c44f14625ffddae1235c7df6)") back than to demonstrate this. You can call it with `cmd1 | magic | cmd2`, and it will find out the names of `cmd1` and `cmd2`. For example:
```
▶ sleep 1 | magic | sleep 2
sleep 1 | magic | sleep 2
```
Unfortunately, this does not work for most practical use cases. It also doesn't work for `crontab -l`. The reason is that typically, if `cmd1/2` is reasonably fast, it will have finished until you can start scanning the `/proc` directory.

**Reply by KaKi87** [reply]
Hello,
Any news on adding `crontab` syntax coloration ?
Thanks

**Reply by keith-hall** [reply]
We haven't got round to it yet, but there's nothing to stop you creating a PR: https://github.com/sharkdp/bat/blob/master/doc/assets.md#adding-new-builtin-languages-for-syntax-highlighting

**Reply by KaKi87** [reply]
No thanks my head blows past the two first steps :sweat_smile:

**Comment by keith-hall** [comment]
https://github.com/sharkdp/bat/pull/2509
